server/oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/utils/AbstractAdapterCredentialsFilter.java

```diff
@@ -184,6 +184,16 @@ protected void logout(Connection loginSession, HttpSession session) {
      */
     abstract protected ConsumerStore createConsumerStore() throws Exception;
 
+    /**
+     * Sets the official servlet URL 
+     * in case this can differ from that in the individual requests.
+     * @see org.eclipse.lyo.server.oauth.core.OAuthConfiguration#setServletUri()
+     * Typical implementation can be:
+     * <pre>return OSLC4JUtils.getServletURI();
+     * </pre>
+     */
+    abstract protected String getServletUri();
```

Comment:
Does it make sense to make the method abstract? Or is it better to provide a default `protected` implementation that does `return OSLC4JUtils.getServletURI();` and could be overridden?

Reply:
I thought about that. There is a risk with giving such a default. it is not sure a server is setting  "OSLC4JUtils.getServletURI()" (Although LyoDesigner always does).
I think not setting the value defaults to NULL, which tells the system to figure out the URL based on the RequestURL. Maybe we can keep this behaviour to be safe?

Reply:
I would really avoid relying on `null` anywhere. The old codebase relied on `null` for default values and it's really bad as it makes it hard to chase NPE bugs etc.

I cannot imagine myself using anything else than `OSLC4JUtils.getServletURI()`, which is why I suggest to provide that by default. The only subclass I could find is [the one from the Bugzilla example](https://bin.garbaye.fr/?bcd77e8d796e1b7c#DriSJHJGVD1GgTKxj4GGV7TarsiH4SWctcvizyVNqMXU) (temp paste until I open-source the projects from 2016-2017). Could you please tell me when would you want something different? I imagine that the signature check would fail otherwise in almost every case.

Again, raising a question why a getter is described as a method that "Sets the official servlet URL" in the Javadoc.

Reply:
@berezovskyi the latest commit now provides a default implementation. Note that this means we have a dependency to lyo-core-settings.